src/main/java/org/hiero/sketch/table/LazyMembership.java

```diff
@@ -41,7 +40,7 @@ public LazyMembership(final IMembershipSet baseMap, final Predicate<Integer> fil
         if (filter == null) throw new NullArgumentException("PartialMembershipDense cannot be " +
                 "instantiated with a null filter");
         this.baseMap = baseMap;
-        this.rowCount = baseMap.getSize(false);
+        this.rowCount = 0;
```

Comment:
I am wondering whether it won't be worthwhile to actually materialize a LazyMembership when you iterate over it.